helix-term/src/ui/text_decorations/diagnostics.rs

```diff
@@ -110,7 +110,7 @@ impl Renderer<'_, '_> {
             (end_col, _) = self.renderer.set_string_truncated(
                 self.renderer.viewport.x + draw_col,
                 row,
-                line,
+                format!("{} {}", self.config.diagnostic_marker, line).as_str(),
```

Comment:
These should just be drawn separately. Allocating a string for every diagnostic on every frame is too expensive

Reply:
That was quick! It makes sense but I am not sure how to do it, would you have any pointers?

Reply:
I guess I would need to change `set_string_truncated` for the `TextRenderer` and the `Surface` or is there a different way that I am missing?

Reply:
He probably means that you should first draw the marker and then draw the line as before. You are allocating here to just draw in a column, which could just be done as offsetting the x value along as you draw.

Reply:
In the end, I changed `set_string_truncated`. The reason that I do not simply call draw_decoration is because, while it is true that without making it configurable is easy:


```rust
// diagnostic marker
const DIAG: &str = "● ";
// trigger before eol diagnostic at <InlineDiagnostics as Decoration>::render_virt_lines
renderer.draw_decoration(DIAG, eol_diagnostic.severity(), virt_off.col as u16);
col_off = renderer.draw_eol_diagnostic(eol_diagnostic, pos.visual_line, virt_off.col);
```

If `DIAG` is not const, and it is part of the `InlineDiagnostics` (state.config), we cannot
make it outlive the `TextRenderer` (I think), so it would not be possible to borrow the diagnostic marker, given this signature:


```rust
impl Decoration for InlineDiagnostics<'_> {
  fn render_virt_lines(
      &mut self,
      renderer: &mut TextRenderer,
      pos: LinePos,
      virt_off: Position,
  ) -> Position {}
}
```

It seems like I am a bit _rusty_, so I might over complicating this.